dashboard/app/models/pd/application/application_base.rb

```diff
@@ -61,12 +61,12 @@ class ApplicationBase < ApplicationRecord
     validate :status_is_valid_for_application_type
     validates_presence_of :type
     validates_presence_of :user_id, unless: proc {|application| application.application_type == PRINCIPAL_APPROVAL_APPLICATION}
-    validates_presence_of :status, unless: proc {|application| application.application_type == PRINCIPAL_APPROVAL_APPLICATION}
```

Comment:
Why is this portion of code being removed? Will this allow both the teacher applicant and the associated admin/principal/school leader to be able to see the status of the same application?

Reply:
This line ensures there is a status for each application record (so both teacher applications and principal approval applications, since both inherit from application_base). It doesn't affect who can see what.

We are already checking `status_is_valid_for_application_type`, so that line covers it because there are statuses defined for all application_base records here https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/pd/application/application_base.rb#L171

(this method is overridden in the teacher application model)

Reply:
I'm guessing this line was here before we had the more specific validation via `status_is_valid_for_application_type`

Reply:
Oh gotcha, thank you!

Reply:
Good question. We are removing a validation -- would that spark new validation errors? I can't imagine it would, but is there a better way to check?

Reply:
It would only cause problems if someone tried to save the old application and it didn't have a `status`. You could query the DB to see if any row is in that situation but if you're reasonably confident there isn't one, that's good enough for me.

Reply:
Ah I see. But if it did not have a status, I would think nothing would be triggered because this validation is now completely removed? It's already passing the other status validation above too.

Reply:
Just ran this on production, so we're good here 👍 
```
[production] dashboard > Pd::Application::ApplicationBase.where(status: nil)
=> #<ActiveRecord::Relation []>
```